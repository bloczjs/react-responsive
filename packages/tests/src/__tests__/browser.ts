import "../App";

import { sizes } from "./sizes.util";

const sleep = (ms: number) =>
  new Promise((res) => setTimeout(res, ms));

const getText = async () => {
  await sleep(100); // ensure that the hooks have time to update
  return (
    (await page.$eval("body", (el) =>
      (el as HTMLElement).innerText
        .replace(/\n/g, "\n\n")
        .replace(/\n\n+/g, "\n\n"),
    )) || ""
  );
};

it("browser test", async () => {
  await page.goto("http://localhost:3000");

  for (const size of sizes) {
    await page.setViewport(size);
    expect(await getText()).toMatchSnapshot();
  }
});
