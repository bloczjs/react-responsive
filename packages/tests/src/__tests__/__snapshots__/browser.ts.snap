// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browser test 1`] = `
"<Only>

Only visible for extra small devices (portrait phones)

Only visible for large or less ([0, 1200px])

Custom component

xs

Only visible for extra small devices (portrait phones)

useBreakpoint()

isXs: true

isXsDown: true

isXsUp: true

isSm: false

isSmDown: true

isSmUp: false

isMd: false

isMdDown: true

isMdUp: false

isLg: false

isLgDown: true

isLgUp: false

isXl: false

isXlDown: true

isXlUp: false

(min-width:768px) and (max-width:992px),(max-width:576px): true

wrong breakpoint: false

wrong media query: false

useBreakpoint() with heights

isThin: true

isThinDown: true

isThinUp: true

isNormal: false

isNormalDown: true

isNormalUp: false

isBig: false

isBigDown: true

isBigUp: false

{

    \\"@media  (max-width:991px)\\": {

        \\"color\\": \\"red\\",

        \\":hover\\": {

            \\"color\\": \\"blue\\"

        }

    },

    \\"@media  (min-width:992px)\\": {

        \\"color\\": \\"green\\"

    }

}"
`;

exports[`browser test 2`] = `
"<Only>

Only visible for small devices (landscape phones)

Only visible for small AND extra large devices

Only visible for small and more ([576px, Infinity[)

Only visible for large or less ([0, 1200px])

Custom component

sm

Only visible for small devices (landscape phones)

sm xl

Only visible for small AND extra large devices

useBreakpoint()

isXs: false

isXsDown: false

isXsUp: true

isSm: true

isSmDown: true

isSmUp: true

isMd: false

isMdDown: true

isMdUp: false

isLg: false

isLgDown: true

isLgUp: false

isXl: false

isXlDown: true

isXlUp: false

(min-width:768px) and (max-width:992px),(max-width:576px): false

wrong breakpoint: false

wrong media query: false

useBreakpoint() with heights

isThin: false

isThinDown: false

isThinUp: true

isNormal: true

isNormalDown: true

isNormalUp: true

isBig: false

isBigDown: true

isBigUp: false

{

    \\"@media  (max-width:991px)\\": {

        \\"color\\": \\"red\\",

        \\":hover\\": {

            \\"color\\": \\"blue\\"

        }

    },

    \\"@media  (min-width:992px)\\": {

        \\"color\\": \\"green\\"

    }

}"
`;

exports[`browser test 3`] = `
"<Only>

Only visible for medium devices (tablets)

Only visible for small and more ([576px, Infinity[)

Only visible for large or less ([0, 1200px])

Custom component

md

Only visible for medium devices (tablets)

useBreakpoint()

isXs: false

isXsDown: false

isXsUp: true

isSm: false

isSmDown: false

isSmUp: true

isMd: true

isMdDown: true

isMdUp: true

isLg: false

isLgDown: true

isLgUp: false

isXl: false

isXlDown: true

isXlUp: false

(min-width:768px) and (max-width:992px),(max-width:576px): true

wrong breakpoint: false

wrong media query: false

useBreakpoint() with heights

isThin: false

isThinDown: false

isThinUp: true

isNormal: true

isNormalDown: true

isNormalUp: true

isBig: false

isBigDown: true

isBigUp: false

{

    \\"@media  (max-width:991px)\\": {

        \\"color\\": \\"red\\",

        \\":hover\\": {

            \\"color\\": \\"blue\\"

        }

    },

    \\"@media  (min-width:992px)\\": {

        \\"color\\": \\"green\\"

    }

}"
`;

exports[`browser test 4`] = `
"<Only>

Only visible for large devices (desktops)

Only visible for small and more ([576px, Infinity[)

Only visible for large or less ([0, 1200px])

Custom component

lg

Only visible for large devices (desktops)

useBreakpoint()

isXs: false

isXsDown: false

isXsUp: true

isSm: false

isSmDown: false

isSmUp: true

isMd: false

isMdDown: false

isMdUp: true

isLg: true

isLgDown: true

isLgUp: true

isXl: false

isXlDown: true

isXlUp: false

(min-width:768px) and (max-width:992px),(max-width:576px): false

wrong breakpoint: false

wrong media query: false

useBreakpoint() with heights

isThin: false

isThinDown: false

isThinUp: true

isNormal: true

isNormalDown: true

isNormalUp: true

isBig: false

isBigDown: true

isBigUp: false

{

    \\"@media  (max-width:991px)\\": {

        \\"color\\": \\"red\\",

        \\":hover\\": {

            \\"color\\": \\"blue\\"

        }

    },

    \\"@media  (min-width:992px)\\": {

        \\"color\\": \\"green\\"

    }

}"
`;

exports[`browser test 5`] = `
"<Only>

Only visible for extra large devices (large desktops)

Only visible for small AND extra large devices

Only visible for small and more ([576px, Infinity[)

Custom component

xl

Only visible for extra large devices (large desktops)

sm xl

Only visible for small AND extra large devices

useBreakpoint()

isXs: false

isXsDown: false

isXsUp: true

isSm: false

isSmDown: false

isSmUp: true

isMd: false

isMdDown: false

isMdUp: true

isLg: false

isLgDown: false

isLgUp: true

isXl: true

isXlDown: true

isXlUp: true

(min-width:768px) and (max-width:992px),(max-width:576px): false

wrong breakpoint: false

wrong media query: false

useBreakpoint() with heights

isThin: false

isThinDown: false

isThinUp: true

isNormal: false

isNormalDown: false

isNormalUp: true

isBig: true

isBigDown: true

isBigUp: true

{

    \\"@media  (max-width:991px)\\": {

        \\"color\\": \\"red\\",

        \\":hover\\": {

            \\"color\\": \\"blue\\"

        }

    },

    \\"@media  (min-width:992px)\\": {

        \\"color\\": \\"green\\"

    }

}"
`;
